File: Frontend/frontend/src/app/models/Show.ts
import { Movie } from './Movie';
import { Room } from './Room';

export class Show {
  id: number;
  movieId: number;
  roomId: number;
  startTime: Date;
  endTime: Date;
  price: number;
  constructor(
    id: number,
    movieId: number,
    roomId: number,
    startTime: Date,
    endTime: Date,
    price: number
  ) {
    this.id = id;
    this.movieId = movieId;
    this.roomId = roomId;
    this.startTime = startTime;
    this.endTime = endTime;
    this.price = price;
  }
}

File: Theater-service/src/main/java/com/theater/entities/Show.java
package com.theater.entities;


import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

import com.fasterxml.jackson.annotation.JsonBackReference;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "shows")
@Builder
public class Show {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "movie_id", nullable = false)
    private Movie movie;


    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "room_id", nullable = false)
    @JsonBackReference
    private Room room;


    @Column(name = "start_time", nullable = false)
    private LocalDateTime startTime;

    @Column(name = "end_time", nullable = false)
    private LocalDateTime endTime;

    @Column(nullable = false)
    private double price;
}
